main/lsp/LSPTypecheckerCoordinator.cc

```diff
@@ -0,0 +1,74 @@
+#include "main/lsp/LSPTypecheckerCoordinator.h"
+
+#include "absl/synchronization/notification.h"
+
+namespace sorbet::realmain::lsp {
+using namespace std;
+
+LSPTypecheckerCoordinator::LSPTypecheckerCoordinator(const shared_ptr<const LSPConfiguration> &config)
+    : typecheckerThreadId(this_thread::get_id()), shouldTerminate(false), typechecker(config), config(config) {}
+
+void LSPTypecheckerCoordinator::asyncRunInternal(function<void()> &&lambda) {
+    if (this_thread::get_id() == typecheckerThreadId) {
```

Comment:
can we check on `sorbet::emscripten_build` instead? or some other more direct way of checking mode

Reply:
What's a more direct way?

We run in single-threaded mode in Emscripten and in tests.

Do you want an explicit `mode` field on the object?

Reply:
Done!